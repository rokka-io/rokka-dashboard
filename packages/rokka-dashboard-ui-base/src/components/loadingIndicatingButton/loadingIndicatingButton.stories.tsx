import { action } from '@storybook/addon-actions';
import { boolean, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';

import { LoadingIndicatingButton } from './loadingIndicatingButton';

storiesOf('Components / Forms', module).add('LoadingIndicatingButton', () => (
  <LoadingIndicatingButton onClick={action('Button Click')} loading={boolean('loading', true)}>
    {text('text', 'foobar')}
  </LoadingIndicatingButton>
));
