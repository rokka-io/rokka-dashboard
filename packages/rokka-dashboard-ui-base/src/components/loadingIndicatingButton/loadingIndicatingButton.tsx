import React from 'react';
import { Button, ButtonProps, Spinner } from '../../elements';
import { colors } from '../../identity/colors/colors';

interface LoadingIndicatingButtonProps extends ButtonProps {
  loading?: boolean;
}

export function LoadingIndicatingButton({ loading, children, ...props }: LoadingIndicatingButtonProps) {
  if (loading) {
    props.disabled = true;
  }

  return <Button {...props}>{loading ? <Spinner small={true} backgroundColor={colors.tints.white} /> : children}</Button>;
}
