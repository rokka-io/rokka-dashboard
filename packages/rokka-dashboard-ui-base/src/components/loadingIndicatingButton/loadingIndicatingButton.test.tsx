import { mount, shallow } from 'enzyme';
import React from 'react';
import { LoadingIndicatingButton } from './loadingIndicatingButton';

describe('Components', () => {
  describe('LoadingIndicatingButton', () => {
    it('renders a button with a loading indicator correctly.', () => {
      const component = shallow(<LoadingIndicatingButton loading={true}>test</LoadingIndicatingButton>);

      expect(component).toMatchSnapshot();
    });
    it('disables onClick when loading is true.', () => {
      const spy = jest.fn();
      const component = mount(
        <LoadingIndicatingButton onClick={spy} loading={true}>
          test
        </LoadingIndicatingButton>
      );

      component.simulate('click');

      expect(spy).toHaveBeenCalledTimes(0);
    });
    it('renders a normal button if loading is false, correctly.', () => {
      const spy = jest.fn();
      const component = shallow(<LoadingIndicatingButton onClick={spy}>test</LoadingIndicatingButton>);

      component.simulate('click');

      expect(component).toMatchSnapshot();
      expect(spy).toHaveBeenCalledTimes(1);
    });
  });
});
