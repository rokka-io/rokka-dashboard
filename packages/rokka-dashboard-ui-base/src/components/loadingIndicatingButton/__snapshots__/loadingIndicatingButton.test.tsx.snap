// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components LoadingIndicatingButton renders a button with a loading indicator correctly. 1`] = `
<Button
  disabled={true}
>
  <Spinner
    backgroundColor="#fff"
    small={true}
  />
</Button>
`;

exports[`Components LoadingIndicatingButton renders a normal button if loading is false, correctly. 1`] = `
<Button
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
>
  test
</Button>
`;
